v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go

```diff
@@ -307,42 +298,115 @@ func (report *ResourceVersionsReport) WriteGroupResourcesReportToBuffer(
 		buffer.WriteString("\n\n")
 	}
 
-	summary := report.createSummary(kinds)
+	return report.writeGroupSections(group, kinds, buffer)
+}
 
-	buffer.WriteString(summary)
-	buffer.WriteString("\n\n")
+func (report *ResourceVersionsReport) writeGroupSections(
+	group string,
+	kinds astmodel.TypeDefinitionSet,
+	buffer *strings.Builder,
+) error {
+	// By default, we treat everything as released
+	releasedResources := kinds
+
+	// Pull out any deprecated resources
+	deprecatedResources := releasedResources.Where(report.isDeprecatedResource)
+	releasedResources = releasedResources.Except(deprecatedResources)
+
+	// Pull out any prerelease resources
+	prereleaseResources := releasedResources.Where(report.isUnreleasedResource)
+	releasedResources = releasedResources.Except(prereleaseResources)
+
+	// First list prerelease reources, if any
+	if len(prereleaseResources) > 0 {
```

Comment:
can consider moving `len(resources) > 0` into the `writeSection` method

Reply:
Fair call. Done.